scripts/WAM2-Layers Source Region Designer.py

```diff
@@ -0,0 +1,288 @@
+#-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+# Define the center point and distance in degrees
+
+#Center point
+center_lat = -2         
+center_lon = -58.75
+    
+distance = 20               #Distance of the square around the center point, in degrees
+
+
+'''
+
+PLEASE NOTE: region ID's (what specific region or regions) to export as a source region, need to be specified within the code itself. 
+A visual representation of the region is generated to show intermediate steps / the final product. 
+
+'''
+
+
+#%%
+
+'''
+
+METHOD 1 - Using the available regions from the 'region mask' python package
+ 
+'''
+
+if selector == 1:
+        
+    #Visualisation of the regions and their ID's --> select region
+    text_kws = dict(
+        bbox=dict(color="none"),
+        path_effects=[pe.withStroke(linewidth=2, foreground="w")],
+        color="#67000d",
+        fontsize=8,
+    )
+    
+    ax = available_regions.plot(
+        projection=ccrs.Robinson(), add_ocean=True, text_kws=text_kws
+    )
+    
+    ax.set_global()   
+    plt.show()
+    
+    
+    #create (a) mask(s) of the required region(s):
+    mask = regionmask.defined_regions.giorgi.mask_3D(longitude, latitude)   #<-- specify available regions source
+    
+    #Specify region ID - what to store
+    region_id = 2                                                           #single mask id
+    region_ids = [1]                                                        #multiple regions in one mask [id-1,id-1,...]
+    ''''!! NOTE REGION_IDS is INDEX of map - 1 !! '''
+    
+    new_mask = mask.sel(region=region_id).values 
+    new_masks = mask.isel(region=region_ids).values
+    
+    #Make WAM2Layers fitting
+    new_mask = np.where(new_mask == False, 0, 1)                            #Replace True and False by 1 and 0's
+    new_masks = np.where(new_masks == False, 0, 1)                          #Replace True and False by 1 and 0's
+    new_masks = np.nanmean(new_masks,axis=0)                                #Multiple masks into 1D array
+    
+    #Plot as a check
+    plt.contourf(longitude,latitude,new_masks)                              #check (switch between mask or masks when needed)
+    plt.show()
+    
+    #Export as source region for WAM2Layers
+    
+    # create xarray dataset
+    data = new_mask   #switch between mask or masks
+    
+    export = xr.Dataset({'source_region': (['latitude', 'longitude'], data.astype(float))})
+        
+    # set coordinates
+    export['longitude'] = ('longitude', longitude)
+    export['latitude'] = ('latitude', latitude)
+    
+    # save to NetCDF file
+    export.to_netcdf(output_dir+'source_region.nc')
+    
+
+#%%
+
+'''
+
+METHOD 2 - Using a shapefile combined with 'geopandas' and the 'region mask' python package
+ 
+'''
+
+if selector == 2:
+    ds = gpd.read_file(shapefile)                    #load shape file
+    
+    #visualisation
+    ds.boundary.plot(color=None, edgecolor="k", linewidth=0.8)      #check
+    plt.show()
```

Comment:
Note that plotting (showing a graphical display) is not always available, especially when you run from the command line. It might be better to save it, or make showing the plot optional